modules/porous_flow/src/materials/PorousFlowPermeabilityKozenyCarman.C

```diff
@@ -0,0 +1,66 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "PorousFlowPermeabilityKozenyCarman.h"
+
+template<>
+InputParameters validParams<PorousFlowPermeabilityKozenyCarman>()
+{
+  InputParameters params = validParams<Material>();
+
+  MooseEnum poroperm_function("kozeny_carman_fd2=0 kozeny_carman_phi0=1", "kozeny_carman_fd2");
+  params.addParam<MooseEnum>("poroperm_function", poroperm_function, "Function relating porosity and permeability. The options are: kozeny_carman_fd2 = f d^2 phi^n/(1-phi)^m (where phi is porosity, f is a scalar constant with typical values 0.01-0.001, and d is grain size). kozeny_carman_phi0 = k0 (1-phi0)^m/phi0^n * phi^n/(1-phi)^m (where phi is porosity, and k0 is the permeability at porosity phi0)");
+  params.addParam<Real>("k0", "The permeability scalar value (usually in m^2) at the reference porosity, required for kozeny_carman_phi0");
+  params.addParam<RealTensorValue>("k_anisotropy", "A tensor to multiply the calculated scalar permeability, in order to obtain anisotropy if required. Defaults to isotropic permeability if not specified.");
+  params.addParam<Real>("phi0", "The reference porosity, required for kozeny_carman_phi0");
```

Comment:
Do you want to do a RangeCheck here (and elsewhere)?

params.addRangeCheckedParam<Real>("phi0", "phi0 > 0 & phi0 < 1", "doco")

(note only one "&" sign)